src/commands/periscope/periscope.ts

```diff
@@ -0,0 +1,163 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+import * as tmpfile from '../utils/tempfile';
+import * as clusters from '../utils/clusters';
+import { getExtensionPath, longRunning } from '../utils/host';
+import {
+    getClusterDiagnosticSettings,
+    chooseStorageAccount,
+    getStorageInfo,
+    prepareAKSPeriscopeDeploymetFile,
+    generateDownloadableLinks,
+    getWebviewContent
+} from './helpers/periscopehelper';
+import { PeriscopeStorage } from './models/storage';
+import AksClusterTreeItem from '../../tree/aksClusterTreeItem';
+import { reporter } from '../utils/reporter';
+
+export default async function periscope(
+    context: IActionContext,
+    target: any
+): Promise<void> {
+    const kubectl = await k8s.extension.kubectl.v1;
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    // ToDo: Generalise this for other feature as wrapper.
+    reporter.sendTelemetryEvent("command", { command: "aks.periscope" });
+
+    if (cloudExplorer.available && kubectl.available) {
+        const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+
+        if (clusterTarget && clusterTarget.cloudName === "Azure" &&
+            clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+            const cluster = clusterTarget.cloudResource as AksClusterTreeItem;
+            const clusterKubeConfig = await clusters.getKubeconfigYaml(cluster);
+
+            if (clusterKubeConfig) {
+                await runAKSPeriscope(cluster, clusterKubeConfig);
+            }
+        } else {
+            vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        }
+    }
+}
+
+async function runAKSPeriscope(
+    cluster: AksClusterTreeItem,
+    clusterKubeConfig: string
+): Promise<void | undefined> {
```

Comment:
I don't think you need `void | undefined` here - your consumer is not expecting a result.  Just go for `Promise<void>` and use plain `return;` statements (without return values).